src/charm.py

```diff
@@ -487,66 +487,94 @@ def _hwcksum_config_matches_pod_config(self) -> bool:
             existing_content = {}
         return existing_content.get("hwcksum") != self._charm_config.enable_hw_checksum
 
+    def _unit_is_non_active_status(self):
+        if not self.unit.is_leader():
+            # NOTE: In cases where leader status is lost before the charm is
+            # finished processing all teardown events, this prevents teardown
+            # event code from running. Luckily, for this charm, none of the
+            # teardown code is necessary to perform if we're removing the
+            # charm.
+            return BlockedStatus("Scaling is not implemented for this charm")
+        try:  # workaround for https://github.com/canonical/operator/issues/736
+            self._charm_config: CharmConfig = CharmConfig.from_charm(charm=self)  # type: ignore[no-redef]  # noqa: E501
+        except CharmConfigInvalidError as exc:
+            return BlockedStatus(exc.msg)
+        if status := self._check_cpu_compatibility():
+            return status
+        if not self._kubernetes_multus.multus_is_available():
+            return BlockedStatus("Multus is not installed or enabled")
+        if not self._bessd_container.can_connect():
+            return WaitingStatus("Waiting for bessd container to be ready")
```

Comment:
I created this PR only to migrate to the CollectStatus event. I'd prefer handling the `ConnectionError` issue in another PR, as it may involve adding different sanity checks.

Reply:
Yeah, we've agreed on the same with Gulsum. I'll bring it up on the daily today.